Modules/vc-module-imageTools/VirtoCommerce.ImageToolsModule.Data/Repositories/ThumbnailRepository.cs

```diff
@@ -0,0 +1,54 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using VirtoCommerce.ImageToolsModule.Data.Models;
+using VirtoCommerce.Platform.Data.Infrastructure;
+
+namespace VirtoCommerce.ImageToolsModule.Data.Repositories
+{
+    public class ThumbnailRepository : DbContextRepositoryBase<ThumbnailDbContext>, IThumbnailRepository
+    {
+        public ThumbnailRepository(ThumbnailDbContext dbContext)
+            : base(dbContext)
+        {
+        }
+
+        public IQueryable<ThumbnailTaskEntity> ThumbnailTasks => DbContext.Set<ThumbnailTaskEntity>();
+
+        public IQueryable<ThumbnailOptionEntity> ThumbnailOptions => DbContext.Set<ThumbnailOptionEntity>();
+
+        public async Task<ThumbnailTaskEntity[]> GetThumbnailTasksByIdsAsync(string[] ids)
+        {
+            return await ThumbnailTasks
+                .Include(t => t.ThumbnailTaskOptions)
+                    .ThenInclude( o => o.ThumbnailOption)
+                .Where(t => ids.Contains(t.Id))
+                .ToArrayAsync();
+        }
+
+        public async Task<ThumbnailOptionEntity[]> GetThumbnailOptionsByIdsAsync(string[] ids)
+        {
+            return await ThumbnailOptions.Where(x => ids.Contains(x.Id)).ToArrayAsync();
+        }
+
+        public async Task RemoveThumbnailTasksByIdsAsync(string[] ids)
+        {
+            var taskEntities = await GetThumbnailTasksByIdsAsync(ids);
+
+            foreach (var taskEntity in taskEntities)
+            {
+                Remove(taskEntity);
```

Comment:
RemoveAsync

Reply:
In ef core there is no implementation RemoveAsync